### PR TITLE
added Anagram

### PR DESCRIPTION
**Pull request by Acentari**
#1640
## **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**

* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Java files are placed inside an existing directory.
* [ ] All filenames are in all uppercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Java naming conventions.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: strings/Anagram.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+            Collections.sort(charsOfS1);
+            Collections.sort(charsForS2);
+
+
+            if (charsOfS1.equals(charsForS2)) {
+                b = true;
+            }
+
+            else {
```
Comment (rbshealy) [inline strings/Anagram.java:45]: use `} else {`  on one line

File: strings/Anagram.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        ArrayList<Character> charsOfS1 = new ArrayList<>();
+        ArrayList<Character> charsForS2 = new ArrayList<>();
+        if (s1.length() == s2.length()) {
+            for (int i = 0; i < s1.length(); i++) {
+                charsOfS1.add(s1.charAt(i));
+                charsForS2.add(s2.charAt(i));
+            }
+
+            Collections.sort(charsOfS1);
+            Collections.sort(charsForS2);
```
Comment (rbshealy) [inline strings/Anagram.java:38]: Do you need to re-assign both ArrayLists, or does this do it for you?

Comment (Acentari) [reply strings/Anagram.java:38]: This does it for me. If the the sorted ArrayLists are the same this means that the 2 Strings have got the same letters

Comment (Acentari) [reply strings/Anagram.java:38]: This does it for me. If the the sorted ArrayLists are the same this means that the 2 Strings have got the same letters

**Comment by rbshealy** [comment]
Thanks for your pull request. I just checked and we already have this algorithm almost exactly the same way you implemented it. I made a request on #1640 for the author to explain what specifically could be improved about this algorithm. If they end up responding you are free to create another PR making changes to the file we already have.

**Pull request closed without merging**